### PR TITLE
Simplify Android Report Broken Site language and point to external link

### PR DESCRIPTION
**Pull request by vadimstruts**
Resolves https://github.com/brave/brave-browser/issues/36483

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Open Brave Shields pop up dialog.
2. Switch off Brave Shields if It is ON.
3. Click "Report broken site".
4. Make sure that It contains the text mentioned here: https://github.com/brave/brave-browser/issues/35683#issuecomment-1917947647, and "`Learn More.`" - part is clickable external link to the Webcompat wiki page

### CODE REVIEW

File: browser/ui/android/strings/android_brave_strings.grd [hunk 1]
```diff
@@ -1619,7 +1619,10 @@ Are you sure you want to do this?
         Let Brave's developers know that this site doesn't work properly with Shields:
       </message>
       <message name="IDS_REPORT_BROKEN_SITE_TEXT_2" desc="Shields text for broken text warning message.">
-        Note: This site address will be submitted with your Brave version number and your IP address(which will not be stored).
+        Thank you for helping make Brave better for all. This report will only send information necessary for us to fix this site. <ph name="LEARN_MORE">%1$s</ph>
```
Comment (mkarolin) [inline browser/ui/android/strings/android_brave_strings.grd:1622]: Maybe "This report will only **contain** information ..."?

Comment (vadimstruts) [reply browser/ui/android/strings/android_brave_strings.grd:1622]: There is the same string for desktop version: https://github.com/brave/brave-core/blob/master/components/resources/brave_components_strings.grd#L454
In that case we should change both?

Comment (rmcfadden3) [reply browser/ui/android/strings/android_brave_strings.grd:1622]: @mkarolin yeah, I'd agree with that wording change. And yeah, @vadimstruts I think it makes sense to change both Android and Desktop, if they're functionally doing the same thing. So:

`Thank you for helping make Brave better for all. This report will only contain information necessary for us to fix this site.`

**Review by deeppandya — APPROVED** [review]
LGTM

File: browser/ui/android/strings/android_brave_strings.grd (hunk 1 above)
Comment (vadimstruts) [reply browser/ui/android/strings/android_brave_strings.grd]: done

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**